mail_activity_board/views/mail_activity_view.xml

```diff
@@ -0,0 +1,216 @@
+        <field name="model">mail.activity</field>
+        <field name="inherit_id" ref="mail.mail_activity_view_search"/>
+        <field name="mode">primary</field>
+        <field name="arch" type="xml">
+            <xpath expr='//field[@name="res_model_id"]' position='before'>
+                <field name="user_id"/>
+                <field name="res_name" string="Origin (opportunity/customer)"/>
+            </xpath>
+
+            <xpath expr='//filter[@name="activities_my"]' position='after'>
+                <filter string="Act. next month" name="activities_month"
+                        domain="[('date_deadline', '&lt;', (context_today()+datetime.timedelta(days=30)).strftime('%Y-%m-%d'))]"
+                        help="Show activities which date deadline is before than 1 month."/>
+                <filter string="Act. next 6 months" name="activities_6_month"
+                        domain="[('date_deadline', '&lt;', (context_today()+datetime.timedelta(days=180)).strftime('%Y-%m-%d'))]"
+                        help="Show activities which date deadline is before than 6 months."/>
+                <separator/>
+            </xpath>
+
+            <xpath expr='//search/group' position='inside'>
+                <filter string="User" domain="[]" groups="sales_team.group_sale_salesman_all_leads" context="{'group_by':'user_id'}"/>
+                <filter string="Origin" context="{'group_by': 'res_model_id'}"/>
+            </xpath>
+
+        </field>
+    </record>
+
+
+<!--
+    ACTION
+-->
+
+    <record model="ir.actions.act_window" id="open_boards_activities_form_tree">
+        <field name="name">ACTIVITIES</field>
+        <field name="res_model">mail.activity</field>
+        <field name="view_type">form</field>
+        <field name="view_mode">tree,form</field>
+        <field name="groups_id" eval="[(4, ref('sales_team.group_sale_salesman'))]"/>
+        <field name="domain">[]</field>
+        <field name="context">{}</field>
+        <field name="view_ids"
+                   eval="[(5, 0, 0),
+                          (0, 0, {'view_mode': 'tree', 'view_id': ref('mail_activity_view_tree')}),
+                          (0, 0, {'view_mode': 'form', 'view_id': ref('mail_activity_view_form_board')}),
+                          (0, 0, {'view_mode': 'kanban', 'view_id': ref('mail_activity_view_kanban')}),
+                          (0, 0, {'view_mode': 'calendar'}),
+                          (0, 0, {'view_mode': 'pivot'}),
+                          (0, 0, {'view_mode': 'graph'})]"/>
+        <field name="search_view_id" ref="mail_activity_view_search"/>
+    </record>
+
+<!--
+Action
+-->
+
+    <record id="action_your_activities" model="ir.actions.server">
+        <field name="name">Boards: My Activities</field>
+        <field name="model_id" ref="model_mail_activity"/>
+        <field name="state">code</field>
+        <field name="code">action = model.action_your_activities()</field>
```

Comment:
I think it's generally better the other way around: declare the activity in XML and then load it from code if ever needed (you'll find most times you don't need it).